roles/etcd/tasks/pre_upgrade.yml

```diff
@@ -52,5 +52,5 @@
     {{ bin_dir }}/etcdctl --no-sync --peers={{ etcd_access_addresses | regex_replace('https','http') }} member list |
     awk -F"[: =]" '{print "{{ bin_dir }}/etcdctl --peers={{ etcd_access_addresses | regex_replace('https','http') }} member update "$1" https:"$7":"$8}' | bash
   run_once: true
-  when: 'etcd_member_list is defined and "http://" in etcd_member_list.stdout'
+  when: 'etcd_member_list.rc != 2 and "http://" in etcd_member_list.stdout'
```

Comment:
We want to run this only if rc == 0. The possible error codes I can think of are:
0 (shows member list)
2 (can't connect)
4 (can't render output because output is in ssl)
127 (command not found)

So this line should be:
` when: 'etcd_member_list.rc == 0 and "http://" in etcd_member_list.stdout' `